### PR TITLE
Convert all .py files to be valid Python

### PR DESCRIPTION
**Pull request by cclauss**
`raise NotImplementedError('Please replace this line with your code...')` signals to the user where they need to put their code.

This approach enables the codebase to be tested with lint tools like flake8, pyflakes, pylint, etc.

### CODE REVIEW

**Comment by cclauss** [comment]
Bump on this PR?

**Comment by cclauss** [comment]
Bump on this PR?

**Comment by donnemartin** [comment]
Hi @cclauss, many of the code samples are intentionally not fully runnable...some are pseudocode or stubbed out.  I feel raising `NotImplementedError` and asking to have them implemented goes beyond that scope.

I'd prefer to use `pass` instead of an exception.  What do you think?

**Comment by cclauss** [comment]
Is this OK or do you want the comment changed or removed altogether?

**Comment by cclauss** [comment]
Done.

**Comment by cclauss** [comment]
Bump?

File: solutions/object_oriented_design/online_chat/online_chat.py [hunk 1]
```diff
@@ -51,8 +71,10 @@ def __init__(self, first_user, second_user):
 
 class GroupChat(Chat):
 
-    def add_user(self, user):  # ...
-    def remove_user(self, user):  # ... 
+    def add_user(self, user):
+        pass
+    def remove_user(self, user):
```
Comment (donnemartin) [inline solutions/object_oriented_design/online_chat/online_chat.py]: Could we add a line break here?

**Comment by cclauss** [comment]
Done.

**Comment by donnemartin** [comment]
@cclauss thank you!

**Pull request merged**